manifests/service.pp

```diff
@@ -35,14 +35,28 @@
     $service_ensure_real = $service_ensure
   }
 
-  service { 'nginx':
-    ensure     => $service_ensure_real,
-    name       => $service_name,
-    enable     => $service_enable,
-    flags      => $service_flags,
-    hasstatus  => true,
-    hasrestart => true,
+  case $::osfamilly {
+    'OpenBSD': {
+      service { 'nginx':
+        ensure     => $service_ensure_real,
+        name       => $service_name,
+        enable     => $service_enable,
+        flags      => '',
```

Comment:
Haven't tested, but this line should be:

 flags => $service_flags,

otherwise looks exactly as what I would have done.